src/utils/entityIndex.ts

```diff
@@ -0,0 +1,80 @@
+namespace Plottable.Utils {
+
+  /**
+   * EntityIndex stores entities and makes them searchable.
+   * Valid entities must be positioned in Cartesian space.
+   */
+  export interface EntityIndex<T extends PositionedEntity> {
```

Comment:
nit: "Index" in the name reminds me of "index in an array"; can we use a different name? Maybe EntityStore, EntityCache?